### PR TITLE
Feature/tracefoss 1389 refactoring part2

### PR DESCRIPTION
**Pull request by ds-mwesener**

### CODE REVIEW

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 1]
```diff
@@ -107,32 +105,33 @@ void testCancelInvestigationSuccessful() {
...
+        when(repository.updateQualityNotificationEntity(investigation)).thenReturn(investigationId);
         when(traceabilityProperties.getBpn()).thenReturn(bpn);
         // When
         investigationsPublisherService.approveInvestigation(investigation);
 
         // Then
-        verify(repository).update(investigation);
-        verify(notificationsService).asyncNotificationExecutor(any());
+        verify(repository).updateQualityNotificationEntity(investigation);
+        // TODO here is missing the discovery object as mock
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:133]: should the TODO get merged in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 2]
```diff
@@ -186,19 +202,20 @@ void testUpdateInvestigation() {
...
         when(traceabilityProperties.getBpn()).thenReturn(bpn);
         // When
         investigationsPublisherService.updateInvestigationPublisher(investigationTestData, status, reason);
 
         // Then
-        Mockito.verify(repository).update(investigationTestData);
-        Mockito.verify(notificationsService, times(1)).asyncNotificationExecutor(any(Notification.class));
+        Mockito.verify(repository).updateQualityNotificationEntity(investigationTestData);
+        // TODO here is missing the discovery object as mock
+        // Mockito.verify(notificationsService, times(1)).asyncNotificationExecutor(any(Notification.class));
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:218]: should the TODO get merged in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 3]
```diff
@@ -252,19 +284,20 @@ void testUpdateInvestigationAccepted() {
...
+        QualityNotification investigationTestData = InvestigationTestDataFactory.createInvestigationTestDataWithNotificationList(QualityNotificationStatus.ACKNOWLEDGED, "recipientBPN", notifications);
         when(traceabilityProperties.getBpn()).thenReturn(bpn);
         // When
         investigationsPublisherService.updateInvestigationPublisher(investigationTestData, status, reason);
 
         // Then
-        Mockito.verify(repository).update(investigationTestData);
-        Mockito.verify(notificationsService, times(1)).asyncNotificationExecutor(any(Notification.class));
+        Mockito.verify(repository).updateQualityNotificationEntity(investigationTestData);
+        // TODO here is missing the discovery object as mock
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:299]: TODO

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 4]
```diff
@@ -318,19 +366,21 @@ void testUpdateInvestigationDeclined() {
...
         when(traceabilityProperties.getBpn()).thenReturn(bpn);
+
         // When
         investigationsPublisherService.updateInvestigationPublisher(investigationTestData, status, reason);
 
         // Then
-        Mockito.verify(repository).update(investigationTestData);
-        Mockito.verify(notificationsService, times(1)).asyncNotificationExecutor(any(Notification.class));
+        Mockito.verify(repository).updateQualityNotificationEntity(investigationTestData);
+        // TODO here is missing the discovery object as mock
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:382]: TODO

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 5]
```diff
@@ -384,19 +449,20 @@ void testUpdateInvestigationClose() {
...
+        QualityNotification investigationTestData = InvestigationTestDataFactory.createInvestigationTestDataWithNotificationList(QualityNotificationStatus.ACCEPTED, "senderBPN", notifications);
         when(traceabilityProperties.getBpn()).thenReturn(bpn);
         // When
         investigationsPublisherService.updateInvestigationPublisher(investigationTestData, status, reason);
 
         // Then
-        Mockito.verify(repository).update(investigationTestData);
-        Mockito.verify(notificationsService, times(1)).asyncNotificationExecutor(any(Notification.class));
+        Mockito.verify(repository).updateQualityNotificationEntity(investigationTestData);
+        // TODO here is missing the discovery object as mock
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:464]: TODO

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/InvestigationTestDataFactory.java [hunk 6]
```diff
@@ -134,33 +160,51 @@ public static Investigation createInvestigationTestData(InvestigationStatus inve
...
+                .investigationStatus(QualityNotificationStatus.ACKNOWLEDGED)
+                .affectedParts(List.of(new QualityNotificationAffectedPart("part123")))
+                .severity(QualityNotificationSeverity.MINOR)
+                .edcNotificationId("123")
+                .messageId("messageId")
+                .isInitial(true)
+                .build();
+        List<QualityNotificationMessage> notifications = List.of(notification);
+
+/*        List<Notification> notifications = List.of(new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/InvestigationTestDataFactory.java:207]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/NotificationTestDataFactory.java [hunk 7]
```diff
@@ -19,19 +19,38 @@
...
+                .description("123")
+                .investigationStatus(QualityNotificationStatus.ACKNOWLEDGED)
+                .affectedParts(affectedParts)
+                .severity(QualityNotificationSeverity.MINOR)
+                .edcNotificationId("123")
+                .targetDate(Instant.parse("2022-03-01T12:00:00Z"))
+                .messageId("messageId")
+                .isInitial(true)
+                .build();
+       /* return new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/NotificationTestDataFactory.java:53]: Should the Comment block stay in?

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/domain/repository/QualityNotificationRepository.java [hunk 8]
```diff
@@ -19,29 +19,35 @@
...
+    long countQualityNotificationEntitiesByStatus(QualityNotificationStatus qualityNotificationStatus);
+
+    long countQualityNotificationEntitiesBySide(QualityNotificationSide investigationSide);
+
+    QualityNotificationId saveQualityNotificationEntity(QualityNotification investigation);
+
+    QualityNotificationId updateQualityNotificationEntity(QualityNotification investigation);
+
+    void updateQualityNotificationMessageEntity(QualityNotificationMessage notification);
+
```
Comment (ds-mmaul) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/domain/repository/QualityNotificationRepository.java:51]: Doesnt the Code speak for itself when doing operations on the class of the repository? My Suggestion would be to minimize the naming addition 'QualityNotification' in the function names. for example  'updateMessage', 'update', 'countBySide', ... . Open for discussion here

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/common/mapper/InvestigationMapperTest.java [hunk 9]
```diff
@@ -55,9 +55,22 @@ void testToReceiverInvestigation() {
...
+                .affectedParts(List.of(new QualityNotificationAffectedPart("123")))
+                .senderManufacturerName("senderManufacturerName")
+                .senderBpnNumber(sender)
+                .receiverBpnNumber(receiver)
+                .receiverManufacturerName("receiverManufacturerName")
+                .severity(QualityNotificationSeverity.MINOR)
+                .isInitial(false)
+                .messageId("1")
+                .build();
+/*        Notification notification = new Notification("1",
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/common/mapper/InvestigationMapperTest.java:71]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/EdcNotificationServiceTest.java [hunk 10]
```diff
@@ -66,7 +65,13 @@ void testNotificationsServiceUpdateAsync() {
...
         when(discoveryService.getDiscoveryByBPN(bpn)).thenReturn(discovery);
         // and
-        Notification notification = new Notification(
+        QualityNotificationMessage notification = QualityNotificationMessage.builder()
+                .receiverBpnNumber(bpn)
+                .targetDate(Instant.now())
+                .severity(QualityNotificationSeverity.MINOR)
+                .isInitial(false)
+                .build();
+      /*  Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/EdcNotificationServiceTest.java:74]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 11]
```diff
@@ -142,11 +141,20 @@ void testUpdateInvestigation() {
...
+        QualityNotificationMessage notification = QualityNotificationMessage.builder()
+                .id("123")
+                .notificationReferenceId("id123")
+                .created(LocalDateTime.now())
+                .targetDate(Instant.now())
+                .investigationStatus(QualityNotificationStatus.CREATED)
+                .affectedParts(affectedParts)
+                .build();
+
+ /*       Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:157]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 12]
```diff
@@ -165,9 +173,17 @@ void testUpdateInvestigation() {
...
-        Notification notification2 = new Notification(
+        QualityNotificationMessage notification2 = QualityNotificationMessage.builder()
+                .id("456")
+                .notificationReferenceId("id123")
+                .created(LocalDateTime.now().plusSeconds(10))
+                .targetDate(Instant.now())
+                .isInitial(false)
+                .build();
+
+      /*  Notification notification2 = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:186]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 13]
```diff
@@ -208,11 +225,19 @@ void testUpdateInvestigationAccepted() {
...
+        List<QualityNotificationAffectedPart> affectedParts = List.of(new QualityNotificationAffectedPart("partId"));
+        QualityNotificationMessage notification = QualityNotificationMessage.builder()
+                .id("123")
+                .notificationReferenceId("id123")
+                .created(LocalDateTime.now())
+                .targetDate(Instant.now())
+                .investigationStatus(QualityNotificationStatus.CREATED)
+                .affectedParts(affectedParts)
+                .build();
+      /*  Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:240]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 14]
```diff
@@ -231,9 +256,16 @@ void testUpdateInvestigationAccepted() {
...
+        );*/
+        QualityNotificationMessage notification2 = QualityNotificationMessage.builder()
+                .id("456")
+                .notificationReferenceId("id123")
+                .created(LocalDateTime.now().plusSeconds(10))
+                .targetDate(Instant.now())
+                .investigationStatus(QualityNotificationStatus.CREATED)
+                .affectedParts(affectedParts)
+                .build();
+       /* Notification notification2 = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:268]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 15]
```diff
@@ -273,12 +306,19 @@ void testUpdateInvestigationDeclined() {
...
+        List<QualityNotificationAffectedPart> affectedParts = List.of(new QualityNotificationAffectedPart("partId"));
+        QualityNotificationMessage notification = QualityNotificationMessage.builder()
+                .id("123")
+                .notificationReferenceId("id123")
+                .created(LocalDateTime.now())
+                .targetDate(Instant.now())
+                .investigationStatus(QualityNotificationStatus.CREATED)
+                .affectedParts(affectedParts)
+                .build();
+       /* Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:321]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 16]
```diff
@@ -297,9 +337,17 @@ void testUpdateInvestigationDeclined() {
...
+
+        QualityNotificationMessage notification2 = QualityNotificationMessage.builder()
+                .id("456")
+                .notificationReferenceId("id123")
+                .investigationStatus(QualityNotificationStatus.CREATED)
+                .affectedParts(affectedParts)
+                .created(LocalDateTime.now().plusSeconds(10))
+                .targetDate(Instant.now())
+                .build();
+      /*  Notification notification2 = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:350]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 17]
```diff
@@ -340,11 +390,26 @@ void testUpdateInvestigationClose() {
...
+                .build();
+
+        QualityNotificationMessage notification2 = QualityNotificationMessage.builder()
+                .id("456")
+                .notificationReferenceId("id123")
+                .created(LocalDateTime.now().plusSeconds(10))
+                .investigationStatus(QualityNotificationStatus.CREATED)
+                .affectedParts(affectedParts)
+                .build();
+      /*  Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:412]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java [hunk 18]
```diff
@@ -406,11 +472,17 @@ void testUpdateInvestigationInvalid() {
...
-        List<AffectedPart> affectedParts = List.of(new AffectedPart("partId"));
-        Notification notification = new Notification(
+        List<QualityNotificationAffectedPart> affectedParts = List.of(new QualityNotificationAffectedPart("partId"));
+        QualityNotificationMessage notification = QualityNotificationMessage.builder()
+                .id("123")
+                .notificationReferenceId("id123")
+                .investigationStatus(QualityNotificationStatus.CREATED)
+                .affectedParts(affectedParts)
+                .build();
+/*        Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsPublisherServiceTest.java:485]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java [hunk 19]
```diff
@@ -95,29 +113,29 @@ void testHandleNotificationReceiveValidSentNotification() {
...
 
     @Test
     @DisplayName("Test testHandleNotificationUpdateValidAcknowledgeNotificationTransition is valid")
     void testHandleNotificationUpdateValidAcknowledgeNotificationTransition() {
 
         // Given
-        List<AffectedPart> affectedParts = List.of(new AffectedPart("partId"));
-        Notification notification = new Notification(
+        List<QualityNotificationAffectedPart> affectedParts = List.of(new QualityNotificationAffectedPart("partId"));
+       /* Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java:138]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java [hunk 20]
```diff
@@ -150,16 +187,16 @@ void testHandleNotificationUpdateValidAcknowledgeNotificationTransition() {
...
 
     @Test
     @DisplayName("Test testHandleNotificationUpdateValidDeclineNotificationTransition is valid")
     void testHandleNotificationUpdateValidDeclineNotificationTransition() {
 
         // Given
-        List<AffectedPart> affectedParts = List.of(new AffectedPart("partId"));
-        Notification notification = new Notification(
+        List<QualityNotificationAffectedPart> affectedParts = List.of(new QualityNotificationAffectedPart("partId"));
+ /*       Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java:199]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java [hunk 21]
```diff
@@ -192,16 +247,16 @@ void testHandleNotificationUpdateValidDeclineNotificationTransition() {
...
 
     @Test
     @DisplayName("Test testHandleNotificationUpdateValidAcknowledgeNotification is valid")
     void testHandleNotificationUpdateValidAcceptedNotificationTransition() {
 
         // Given
-        List<AffectedPart> affectedParts = List.of(new AffectedPart("partId"));
-        Notification notification = new Notification(
+        List<QualityNotificationAffectedPart> affectedParts = List.of(new QualityNotificationAffectedPart("partId"));
+/*        Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java:259]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java [hunk 22]
```diff
@@ -234,16 +307,16 @@ void testHandleNotificationUpdateValidAcceptedNotificationTransition() {
...
 
     @Test
     @DisplayName("Test testHandleNotificationUpdateValidAcknowledgeNotification is valid")
     void testHandleNotificationUpdateValidCloseNotificationTransition() {
 
         // Given
-        List<AffectedPart> affectedParts = List.of(new AffectedPart("partId"));
-        Notification notification = new Notification(
+        List<QualityNotificationAffectedPart> affectedParts = List.of(new QualityNotificationAffectedPart("partId"));
+ /*       Notification notification = new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/qualitynotification/domain/service/InvestigationsReceiverServiceTest.java:319]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/InvestigationTestDataFactory.java [hunk 23]
```diff
@@ -20,28 +20,46 @@
...
+                .investigationStatus(investigationStatus)
+                .affectedParts(List.of(new QualityNotificationAffectedPart("part123")))
+                .targetDate(Instant.now())
+                .severity(QualityNotificationSeverity.MINOR)
+                .edcNotificationId("1")
+                .messageId("messageId")
+                .isInitial(true)
+                .build();
+        List<QualityNotificationMessage> notifications = List.of(notification);
+/*
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/InvestigationTestDataFactory.java:62]: Should the Comment block stay in?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/InvestigationTestDataFactory.java [hunk 24]
```diff
@@ -61,61 +79,69 @@ public static Investigation createInvestigationTestData(InvestigationStatus inve
...
+                .investigationStatus(notificationInvestigationStatus)
+                .affectedParts(List.of(new QualityNotificationAffectedPart("part123")))
+                .severity(QualityNotificationSeverity.MINOR)
+                .edcNotificationId("123")
+                .messageId("messageId")
+                .isInitial(true)
+                .build();
+        List<QualityNotificationMessage> notifications = List.of(notification);
+
+/*		List<Notification> notifications = List.of(new Notification(
```
Comment (ds-mmaul) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/testdata/InvestigationTestDataFactory.java:144]: Should the Comment block stay in?

**Review by ds-mmaul — CHANGES_REQUESTED** [review]
Please address the comments on the changes

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/application/service/QualityNotificationService.java [hunk 25]
```diff
@@ -28,14 +27,15 @@
 import java.time.Instant;
 import java.util.List;
 
+// todo do not return response return domain
```
Comment (ds-mmaul) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/application/service/QualityNotificationService.java:30]: TODO

**Review by ds-mmaul — CHANGES_REQUESTED** [review]

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/domain/repository/QualityNotificationRepository.java (hunk 8 above)
Comment (ds-mwesener) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/qualitynotification/domain/repository/QualityNotificationRepository.java:51]: Usually I would agree. Currently this is only a preperation for quality alerts and will not stay as it is. Each method will be moved to the specific implementation. Only commons will stay here. But good catch!

**Review by ds-mmaul — APPROVED** [review]
LGTM

**Pull request merged**